cpp/include/cudf/strings/char_types/char_types.hpp

```diff
@@ -100,6 +101,50 @@ std::unique_ptr<column> all_characters_of_type(
   string_character_types verify_types = string_character_types::ALL_TYPES,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Filter specific character types from a column of strings.
+ *
+ * To remove all characters of a specific type, set that type in
+ * `types_to_filter` and set `types_to_keep` to `ALL_TYPES`.
+ *
+ * To filter out characters NOT of a select type, specify `ALL_TYPES` for
+ * `types_to_filter` and which types to not remove in `types_to_keep`.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ['ab', 'a b', 'a7bb', 'A7B234']
+ * s1 = s.filter_characters_of_type(s,NUMERIC,"",ALL_TYPES)
+ * s1 is ['ab', 'a b', 'abb', 'AB']
+ * s2 = s.filter_characters_of_type(s,ALL_TYPES,"-",LOWER)
+ * s2 is ['ab', 'a-b', 'a-bb', '------']
+ * @endcode
+ *
+ * In `s1` all NUMERIC types have been removed.
+ * In `s2` all non-LOWER types have been replaced.
+ *
+ * One but not both parameters `types_to_filter` and `types_to_keep` must
+ * be set to `ALL_TYPES`.
```

Comment:
My intention was to make this more flexible by allowing you to mix types you wanted to keep and remove at the same time. Since that complication was not necessary for the task at hand, I thought I would leave the door open for such possible cases in the future.